Native build fails if quarkus.package.output-directory is set

### Describe the bug

Im my project, we generate several native executables using several executions with several quarkus.package.output-directory.
Since Quarkus 3, each native generation fails if quarkus.package.output-directory is set.


### Expected behavior

We expect to have as many binares as executions nodes in quarkus-maven-plugin.

### Actual behavior

The native executable generation fails with :
```[ERROR] Failed to execute goal io.quarkus.platform:quarkus-maven-plugin:3.0.3.Final:build (oracle) on project getting-started: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR] 	[error]: Build step io.quarkus.deployment.pkg.steps.JarResultBuildStep#buildNativeImageJar threw an exception: java.io.UncheckedIOException: Unable to copy json config file from /resources-config.json to /home/osboxes/Documents/Formations/quarkus/getting-started/target/oracle-quarkus-app/getting-started-1.0.0-SNAPSHOT-native-image-source-jar
```

In our real project we also have reflection-config.json and other properties with just this file it fails



### How to Reproduce?

Here is a little project done with getting-started quarkus project to rerpduce.
[getting-started.zip](https://github.com/quarkusio/quarkus/files/11500033/getting-started.zip)

### Output of `uname -a` or `ver`

Linux localhost.localdomain 6.2.14-300.fc38.x86_64 #1 SMP PREEMPT_DYNAMIC Mon May  1 00:55:28 UTC 2023 x86_64 GNU/Linux

### Output of `java -version`

Java HotSpot(TM) 64-Bit Server VM (build 17.0.1+12-LTS-39, mixed mode, sharing)

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

3.0.3.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.2 (ea98e05a04480131370aa0c110b8c54cf726c06f)

### Additional information

_No response_

--- a/core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/JarResultBuildStep.java
+++ b/core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/JarResultBuildStep.java
@@ -1030,6 +1030,7 @@ public class JarResultBuildStep {
                     @Override
                     public void accept(Path jsonPath) {
                         try {
+                            Files.createDirectories(thinJarDirectory);
                             Files.copy(jsonPath, thinJarDirectory.resolve(jsonPath.getFileName().toString()));
                         } catch (IOException e) {
                             throw new UncheckedIOException(

